server/influx_test.go

```diff
@@ -217,6 +217,104 @@ func TestService_Influx_UseCommand(t *testing.T) {
 	}
 }
 
+// TestService_Influx_CommandWithOnClause tests preprocessing of command with ON clause
+func TestService_Influx_CommandWithOnClause(t *testing.T) {
+	tests := []struct {
+		name string
+		db   string
+		rp   string
+	}{
+		{
+			name: "/* no command */",
+		},
+		{
+			name: "SHOW MEASUREMENTS",
+		},
+		{
+			name: "SHOW TAG KEYS ON mydb",
+			db:   "mydb",
+		},
+		{
+			name: "SHOW TAG KEYS ON mydb FROM table",
+			db:   "mydb",
+		},
+		{
+			name: "USE anotherdb; SHOW TAG KEYS ON mydb",
+			db:   "anotherdb",
+		},
+		{
+			name: `show tag keys on "mydb"`,
+			db:   "mydb",
+		},
```

Comment:
We should also cover the typo error syntax such as: `oN`:

```suggestion
		{
			name: `show tag keys on "mydb"`,
			db:   "mydb",
		},
		{
			name: `show tag keys oN "mydb"`,
			db:   "mydb",
		},
```